projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionService.java

```diff
@@ -373,25 +373,24 @@ public void syncJobExecutionStatuses(List<String> runningJobExecutionIds) {
     if (runningJobExecutionIds == null) {
       return;
     }
-
+    var runningJobStatus = List.of(ExecutionStatus.SUBMITTED, ExecutionStatus.RUNNING);
     List<com.vmware.taurus.service.model.DataJobExecution> dataJobExecutionsToBeUpdated =
         jobExecutionRepository
             .findDataJobExecutionsByStatusInAndStartTimeBefore(
```

Comment:
Why priorly selecting the entire entity, to then only filter by its id for the subsequent update query?

Reply:
@mrMoZ1

Reply:
Created a new pull request addressing your comment: https://github.com/vmware/versatile-data-kit/pull/1278